CMakeLists.txt

```diff
@@ -77,9 +77,29 @@ if(ENABLE_OC AND CMAKE_SIZEOF_VOID_P EQUAL 8 AND NOT WIN32)
       list(APPEND EOSIO_WASM_RUNTIMES eos-vm-oc)
       # EOS VM OC requires LLVM, but move the check up here to a central location so that the EosioTester.cmakes
       # can be created with the exact version found
-      find_package(LLVM REQUIRED CONFIG)
+      # find_package VERSION range is only available in 3.19 and newer.
+      if(${CMAKE_VERSION} VERSION_LESS "3.19")
+         find_package(LLVM REQUIRED CONFIG)
+      else()
+         # Preferring the latest LLVM version, search through major 11..7 and minor 9..0 in reverse order.
+         # Minor versions may break API. This manifests in the LLVM config files such that a range of `11...<12`
+         # will NOT find LLVM `11.1`. We loop over minor as well as major to resolve this.
+         set(max_minor 9)
+         set(major 11)
+         set(minor ${max_minor})
+         while(NOT LLVM_FOUND AND major GREATER_EQUAL 7)
+            math(EXPR end "${major} + 1")
+            find_package(LLVM ${major}.${minor}...<${end} CONFIG QUIET)
+            if(minor GREATER 0)
+               math(EXPR minor "${minor} - 1")
+            else()
+               math(EXPR major "${major} - 1")
+               set(minor ${max_minor})
+            endif()
+         endwhile()
+      endif()
       if(LLVM_VERSION_MAJOR VERSION_LESS 7 OR LLVM_VERSION_MAJOR VERSION_GREATER_EQUAL 12)
-	      message(FATAL_ERROR "Leap requires an LLVM version 7 through 11")
+	 message(FATAL_ERROR "Leap requires an LLVM version 7 through 11")
```

Comment:
is there a tab here? otherwise LGTM. Consider changing the README to remove the `CMAKE_PREFIX_PATH`
https://github.com/AntelopeIO/leap/blob/8bf37aac8ab402a328035ca9f321f43e5097fc54/README.md?plain=1#L149

Reply:
It looks like there was and still is a tab on this line. Pretty weird, wonder how it got there. Might as well remove it while we're here.

![shot-2023-03-10_09-45-18](https://user-images.githubusercontent.com/32485495/224345670-b7bab3f7-7586-4325-a0e8-032b69c54046.png)

Reply:
So there is. I seem to have removed the automatic untabify from my emacs config. I've untabified and pushed the change.